### ISSUE TITLE
Validate geneset information

### ISSUE THREAD

**Issue by MDunitz** [chanzuckerberg/corpora-data-portal#909]
## Approach: 

See the format and validation requirements in [Curators can upload gene sets to a private collection](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell/115)

~- Validate the following conditions for each gene set
    - Missing Gene Set Name
    - Duplicate Gene Set Name (within the file)
    - Duplicate Gene set name (already in the db)- [This will require updating the /collection/{collection_uuid}/ api to also return geneset information]
    - Missing Gene Set Description
    - No Gene Symbols in Gene Set (The set is empty)
    - Duplicate Gene Symbols in Gene Set~
## DOD:
- Create test gene set files containing multiple gene sets representing the validation errors above
- Test that all of the validation errors are caught and the error response specifies the gene set name and the validation error
- All validation tests should be run each time (don't stop when one fails to ensure uploading gene sets isn't an endless nightmare of errors)

==== TEAM COMMENTS ====
Timmy: Signe and Explorer Team are creating test files for geneset validation. Data Portal will need to modify the test files to meet the stricter validation rules DP imposes!

**Comment by tihuan** [comment]
Repointed from 5 to 8 for the following reasons:

1. Checking for CSV encoding type as unicode (UTF-X instead of, say, BIG 5) turned out to require jumping through more hoops than expected in code with research involved in figuring out the best approach and comparing different libraries

1. Originally I was thinking we could just write an easy function to loop through the CSV rows, but I forgot that we had to deal with the possibility of weirdly formed CSVs that have comments, different delimiter, and potential performance issue with thousands of rows. So there's more logics to write and research to do to find the best library to build a robust solution on top of

1. Some extra clarification had to be done to completely understand the validation rules. And some requirements (such as validating no duplicate gene set name in CSV file) led to introducing more complexity to the validation strategy that I had to refactor the code to take them into account

That's it!
